libs/platform/kernel/ebpf_platform_kernel.c

```diff
@@ -761,6 +761,32 @@ ebpf_platform_thread_id()
     return (uint32_t)(uintptr_t)PsGetCurrentThreadId();
 }
 
+_IRQL_requires_max_(PASSIVE_LEVEL) _Must_inspect_result_ ebpf_result_t
+    ebpf_platform_get_authentication_id(_Out_ uint64_t* authentication_id)
+{
+    SECURITY_SUBJECT_CONTEXT context = {0};
+    SeCaptureSubjectContext(&context);
+    LUID local_authentication_id;
+
+    PACCESS_TOKEN access_token = SeQuerySubjectContextToken(&context);
+    // SeQuerySubjectContextToken() is not expected to fail.
+    ebpf_assert(access_token != NULL);
+    if (access_token == NULL) {
+        return EBPF_FAILED;
```

Comment:
why dont we use the bail_on_error pattern, and return_ebpf_error so that the failure code can be traced?